src/modules/visits/hooks/get-visit-access-status.ts

```diff
@@ -1,12 +1,28 @@
-import { useMutation, UseMutationResult } from 'react-query';
+import { useMutation, UseMutationResult, useQuery, UseQueryResult } from 'react-query';
 
+import { QUERY_KEYS } from '@shared/const';
 import { VisitAccessStatus } from '@shared/types';
 import { VisitsService } from '@visits/services';
 
-export function useGetVisitAccessStatus(): UseMutationResult<
+// Query
+
+export function useGetVisitAccessStatus(
+	slug: string,
+	enabled = true
+): UseQueryResult<VisitAccessStatus | null> {
+	return useQuery(
+		[QUERY_KEYS.getVisitAccessStatus, { slug }],
+		() => VisitsService.getAccessStatusBySpaceSlug(slug),
+		{ enabled }
+	);
+}
```

Comment:
Since the enabled currently only depends on the slug that you already pass to the function, you can just use slug to determine the enabled inside the function.
```suggestion
export function useGetVisitAccessStatus(
	slug: string | undefined
): UseQueryResult<VisitAccessStatus | null> {
	return useQuery(
		[QUERY_KEYS.getVisitAccessStatus, { slug }],
		() => VisitsService.getAccessStatusBySpaceSlug(slug),
		{ enabled: typeof slug === 'string' }
	);
}
```

Then you can simplify the code to: 
```
	const { data: accessStatus } = useGetVisitAccessStatus(slug);
```

if the enabled boolean depends on other variables that are not passed to the function, you can still use the enabled parameter.